walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,106 +41,223 @@ public int compare(Event a, Event b) {
         }
       };
 
-  /**
-   * Returns a list of time periods in which the meeting, specified by request, could happen. If one
-   * or more time slots exists so that both mandatory and optional attendees can attend, it returns
-   * those time slots. Otherwise, it returns the time slots that fit just the mandatory attendees.
-   *
-   * @param eventsCollection the events we know about
-   * @param request information about the meeting, including attendees, optional attendees, and how
-   *     long it needs to be
-   */
-  public Collection<TimeRange> query(Collection<Event> eventsCollection, MeetingRequest request) {
-    Collection<TimeRange> withOptionalAttendees = getMeetingTimes(eventsCollection, request, true);
-
-    // Special case: if no mandatory attendees and optional attendees' schedules cannot fit in a
-    // meeting, no meeting times are possible.
-    return !withOptionalAttendees.isEmpty() || request.getAttendees().isEmpty()
-        ? withOptionalAttendees
-        : getMeetingTimes(eventsCollection, request, false);
-  }
+  // All logic needed to find optimal meeting times with one instance of meeting information and
+  // events to be considered.
+  private static class SingleMeetingResolver {
+    // all events to be considered
+    private final Collection<Event> events;
+
+    // information about the meeting, including attendees, optional attendees, and how long it needs
+    // to be
+    private final MeetingRequest request;
+
+    private ArrayList<TimeRange> optimalMeetingTimes = new ArrayList<TimeRange>();
+
+    // best attendance of optional attendees seen so far
+    private int bestAttendance;
 
-  private Collection<TimeRange> getMeetingTimes(
-      Collection<Event> eventsCollection,
-      MeetingRequest request,
-      boolean includeOptionalAttendees) {
-    HashSet<String> attendees = new HashSet<String>(request.getAttendees());
-    if (includeOptionalAttendees) {
-      attendees.addAll(request.getOptionalAttendees());
+    // attendance optional attendees of current meeting
+    private int currAttendance;
+
+    // minimum meeting time
+    private long minTime;
```

Comment:
Call this "minMeetingDuration". You want to be crisp when you're talking about a time (a specific moment in the universe) or a duration (a span between two points in time).

"minTime" give the reader zero intuition about the meaning of the variable at usage sites, and given the above note on time vs duration), is actually misleading.